helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -705,6 +715,8 @@ private GenericHelixController(PipelineRegistry registry, PipelineRegistry taskR
 
     _onDemandRebalanceTimer =
         new Timer("GenericHelixController_" + _clusterName + "_onDemand_Timer", true);
+    _participantDeregistrationTimer
+        = new Timer("GenericHelixController_" + _clusterName + "_participantDeregistration_Timer", true);
```

Comment:
Since it was a same pipeline, I do suggest to leverage the same timer. This operation also counts as the ondemand rebalancing.

Reply:
Hi @junkaixue , thanks for the review! Just want to clarify your feedback:
(1) Move the `ParticipantDeregistrationStage` to the end of the rebalance pipeline - currently it belongs to a new, separate pipeline = `ParticipantDeregistrationPipeline`
(2) Leverage `scheduleOnDemandRebalance` as participant deregistration now occurs as part of onDemand rebalance

Reply:
Yes.